contents/blog/persons-on-events.md

```diff
@@ -0,0 +1,39 @@
+---
+date: 2022-11-02
+title: 'How we’re improving performance by combining persons and events’'
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+categories: ['Release notes', 'Product updates']
+featuredImage: ../images/blog/posthog-engineering-blog.png
+featuredImageType: full
+---
+
+In [a previous product update](/blog/the-posthog-array-1-39-0) we announced a beta for a substantial change to the way we handle persons and events on PostHog. Today, after gathering your feedback and seeing the improvements to query performance, we’re rolling this change out to all PostHog Cloud users. We've also made this change available for Self-Hosted users, as part of the 1.41.0 update. 
+
+> **Need to upgrade a self-hosted instance?** We've explained how to upgrade to 1.41.0 and run the necessary async migrations in [our 1.41.0 release highlights](/blog/the-posthog-array-1-41-0). 
+
+This change combines [persons](/manual/persons) and [events](/manual/events) into a single ClickHouse table, adding person IDs and properties _onto_ events. This change also applies to groups, adding group properties and aggregation to the same table. 
+
+You won’t see any UI changes — persons will still have their own Persons & Groups section on the sidebar, for example — and there’ll be no loss of functionality, but here are some changes you may see:
+
+- **Faster results for queries involving person or group properties and events**. Putting persons, groups and events into a single ClickHouse table means we no longer have to join tables to get results on queries involving these data. As a result, query performance will improve by up to 400%.
+
+- **Faster filtering of events with person or group properties**. Similarly, filtering events by person or group properties is much faster when a join is no longer necessary. Anywhere where you’re working with persons or groups, and events, PostHog will be faster. 
+
+- **Users will no longer be merged retroactively in some situations**. When an identified user logs in from a different browser (thus becoming anonymous) we end up with separate records for their unidentified and identified behavior, which are then merged into a single person. We used to do this using database joins at query time, so that all events were tied to the same person. Now, we simply look at the events when queries are created - meaning that anonymous events are shown as separate, unique persons in insights.
```

Comment:
This is not accurate:
```suggestion
- **Users will no longer be merged retroactively in some situations**. When an identified user logs in from a different browser (thus becoming anonymous) we end up with separate records for their unidentified and identified behavior, which are then merged into a single person. We used to do database joins at query time, so that all events were tied to the same person. Now, we simply look at the events which have the data from the event processing time - meaning that anonymous events are shown as separate, unique persons in insights.
```

Alternatively:
**Anonymous part of the sessions can in some cases be seen as belonging to a separate user**. When an identified user logs in from a different browser (thus becoming anonymous) we end up with separate users until the point when they are merged (via identify or alias). Previously at query time we used join's, which meant that if users were ever merge we would associate all the events prior to the merge with the user too. Now at the event time we add the person and groups info to the event. Then at query time we use that data from the event, which means that the anonymous part of the session will look like it belongs to a separate user compared to the logged in user.